### ISSUE TITLE
With vue router

### ISSUE THREAD

**Issue by oscarteg** [igeligel/vuex-feature-scoped-structure#1]
When using vue router with this structure you need to check if the module exists in the create of that module. `this.$store.registerModule('$_products', store);` needs to be replaced with 

```
if (!(this.$store && this.$store.state && this.$store.state.$_products)) {
  this.$store.registerModule('$_products', store);
}
```

Or else the module will be registered every time the page is being called and you get some weird behaviour.

**Comment by igeligel** [comment]
This issue is described here: https://github.com/vuejs/vuex/issues/833 

Would be nice if you could upvote the issue so it gets more awareness

**Comment by igeligel** [comment]
This is fixed with 428b98d52f1fe9d0f255dc1db66f6ad37447848e
